dashboard/app/controllers/scripts_controller.rb

```diff
@@ -72,9 +73,14 @@ def show
 
     # Attempt to redirect user if we think they ended up on the wrong unit overview page.
     override_redirect = VersionRedirectOverrider.override_unit_redirect?(session, @script)
-    if !override_redirect && redirect_unit = redirect_unit(@script, request.locale, @course)
-      redirect_to script_path(redirect_unit) + "?redirect_warning=true"
-      return
+    if !override_redirect && redirect_info = redirect_unit(@script, request.locale, @course)
```

Comment:
how about renaming `redirect_unit` to `redirect_info` or `get_redirect_info`? otherwise its a little confusing to get back something that is not a unit